File: spring-web-java/src/main/java/spring/web/java/controller/MemberController.java
package spring.web.java.controller;

import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.bind.annotation.*;
import spring.web.java.dto.request.MemberSaveRequest;
import spring.web.java.dto.request.MemberUpdateRequest;
import spring.web.java.dto.response.ApiResponse;
import spring.web.java.dto.response.MemberResponse;
import spring.web.java.service.MemberService;

@RestController
@RequestMapping("member")
@RequiredArgsConstructor
public class MemberController {

    private final MemberService memberService;

    @PostMapping
    public ApiResponse<MemberResponse> saveMember(@RequestBody @Valid MemberSaveRequest memberSaveRequest) {
        MemberResponse responseDto = memberService.saveMember(memberSaveRequest);

        return new ApiResponse<>(responseDto);
    }

    @GetMapping
    public ApiResponse<Page<MemberResponse>> findMembers(
            Pageable pageable,
            @RequestParam(required = false) String account,
            @RequestParam(required = false) String name
    ) {
        Page<MemberResponse> members = memberService.findMembers(pageable, account, name);

        return new ApiResponse<>(members);
    }

    @GetMapping("{id}")
    public ApiResponse<MemberResponse> findMember(@PathVariable Long id) {
        MemberResponse responseDto = memberService.findMember(id);

        return new ApiResponse<>(responseDto);
    }

    @PatchMapping("{id}")
    public ApiResponse<MemberResponse> updateMember(
            @PathVariable Long id,
            @RequestBody @Valid MemberUpdateRequest memberUpdateRequest
    ) {
        MemberResponse responseDto = memberService.updateMember(id, memberUpdateRequest);

        return new ApiResponse<>(responseDto);
    }

    @DeleteMapping("{id}")
    public ApiResponse<Void> deleteMember(@PathVariable Long id) {
        memberService.deleteMember(id);

        return new ApiResponse<>();
    }
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/controller/MemberController.kt
package spring.web.kotlin.controller

import org.springframework.data.domain.Pageable
import org.springframework.web.bind.annotation.*
import spring.web.kotlin.dto.request.MemberSaveRequest
import spring.web.kotlin.dto.request.MemberUpdateRequest
import spring.web.kotlin.dto.response.ApiResponse
import spring.web.kotlin.service.MemberService

@RestController
@RequestMapping("member")
class MemberController(
    val memberService: MemberService
) {
    @PostMapping
    fun saveMember(@RequestBody memberSaveRequest: MemberSaveRequest) =
        ApiResponse(memberService.saveMember(memberSaveRequest))

    @GetMapping
    fun findMembers(
        pageable: Pageable,
        @RequestParam(required = false) account: String?,
        @RequestParam(required = false) name: String?
    ) = ApiResponse(memberService.findMembers(pageable, account, name))

    @GetMapping("{id}")
    fun findMember(@PathVariable id: Long) = ApiResponse(memberService.findMember(id))

    @PatchMapping("{id}")
    fun updateMember(
        @PathVariable id: Long,
        @RequestBody memberUpdateRequest: MemberUpdateRequest
    ) = ApiResponse(memberService.updateMember(id, memberUpdateRequest))

    @DeleteMapping("{id}")
    fun deleteMember(@PathVariable id: Long): ApiResponse<Unit> {
        memberService.deleteMember(id)

        return ApiResponse()
    }
}